### ISSUE TITLE
Missing bindings of sys/syscall.h

### ISSUE THREAD

**Issue by marmistrz** [rust-lang/libc#662]
At least on glibc 2.25 there's a header `sys/syscall.h` which defines all the syscall numbers by name. It effectively includes `asm/unistd_64.h` (on x86_64) which is structured like this:

```
#define __NR_read 0
#define __NR_write 1
#define __NR_open 2
```
and `bits/syscall.h`, which does something like this:
```
#define SYS_write __NR_write
```
I don't insist on 1-1 binding but I believe we should have a syscall registry in Rust. Currently, every Rust application that uses `seccomp` or `ptrace` has to hardcode the syscall numbers.

Btw. do you think that such thing should be a part of the `libc` crate or rather an external crate such as https://github.com/nix-rust/nix ?

**Comment by alexcrichton** [comment]
PRs are of course always welcome! There's no reason these haven't been added other than no one's gotten around to doing it yet.

**Comment by marmistrz** [comment]
So let's find a cute name for that... :)
What about `libc::syscall_num::write` for getting the number of `write` syscall?

**Comment by alexcrichton** [comment]
Oh we should just stick to the libc names, not invent new names.

**Comment by marmistrz** [comment]
Ok, so which of two names: `SYS_write` or `__NR_write`? Both of them are available, as mentioned in the first post.

**Comment by marmistrz** [comment]
Closed by #672
